File: ruby-motion-tweetr/app/tweet.rb
class Tweet
  attr_accessor :avatar

  def initialize(data)
    @data = data
  end

  def tweet_text 
    @data["text"]
  end

  def username 
    @data["from_user_name"]
  end

  def avatar_url
    @data["profile_image_url"]
  end

  def method_missing(meth, *args, &blk)
    return @data[meth.to_s] if @data.key?(meth.to_s)
    super
  end
end

File: objective-c-tweetr/tweetr/Tweet.m
//
//  Tweet.m
//  tweetr
//
//  Created by Addam Hardy on 5/20/13.
//
//

#import "Tweet.h"

@implementation Tweet

- (void)initWithData:(NSDictionary *)data
{
    self.tweet = [data objectForKey:@"text"];
    self.username = [data objectForKey:@"from_user_name"];
    self.avatar = [data objectForKey:@"profile_image_url"];
}

@end
